alvr/adb/src/lib.rs

```diff
@@ -0,0 +1,364 @@
+// https://android.googlesource.com/platform/packages/modules/adb/+/refs/heads/main/docs/user/adb.1.md
+
+pub mod connection_state;
+pub mod device;
+pub mod forwarded_port;
+pub mod transport_type;
+
+use std::{collections::HashSet, io::Cursor, path::PathBuf, process::Command, time::Duration};
+
+#[cfg(windows)]
+use std::os::windows::process::CommandExt;
+
+use alvr_common::debug;
+use alvr_filesystem::Layout;
+use anyhow::{anyhow, Context, Result};
+use device::Device;
+use forwarded_port::ForwardedPort;
+use zip::ZipArchive;
+
+// https://developer.android.com/tools/releases/platform-tools#revisions
+// NOTE: At the time of writing this comment, the revisions section above
+// shows the latest version as 35.0.2, but the latest that can be downloaded
+// by specifying a version is 35.0.0
+const PLATFORM_TOOLS_VERSION: &str = "-latest"; // E.g. "_r35.0.0"
+
+#[cfg(target_os = "linux")]
+const PLATFORM_TOOLS_OS: &str = "linux";
+#[cfg(target_os = "macos")]
+const PLATFORM_TOOLS_OS: &str = "darwin";
+#[cfg(windows)]
+const PLATFORM_TOOLS_OS: &str = "windows";
+
+const REQUEST_TIMEOUT: Duration = Duration::from_secs(5);
+
+pub enum WiredConnectionStatus {
+    NotReady(String),
+    Ready,
+}
+
+pub fn setup_wired_connection(
+    layout: &Layout,
+    control_port: u16,
+    stream_port: u16,
+) -> Result<WiredConnectionStatus> {
+    let adb_path = require_adb(layout, |downloaded, total| {
+        let total_display = match total {
+            Some(t) => t.to_string(),
+            None => "?".to_owned(),
+        };
+        debug!("Downloading ADB: got {downloaded} bytes of {total_display}");
```

Comment:
I think it was better to keep `dbg_connection()`. The long term goal would be to remove all plain debug log and replace it with specific debug groups.